### DISCUSSION TITLE
Can not allways commit and sync to repository

### DISCUSSION THREAD

**Discussion by Johan1974** [community/community#63685] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Hello,

I have visual studio code on my windows machine connected to my remote linux environment. Also i have created a github repositiry to save my souce filed. For this i use code spaces. Now when i try to commit and sync i sometimes get an error like this

```
> git add -A -- .
bot01/user_data/hyperopts/sample_hyperopt_loss.py: Permission denied
bot01/user_data/notebooks/strategy_analysis_example.ipynb: Permission denied
error: insufficient permission for adding an object to repository database .git/objects
error: strategies/AwesomeMacd.py: failed to insert into database
error: unable to index file 'strategies/AwesomeMacd.py'
fatal: updating files failed
```

After i exceute the sam command with sudo i can commit and sync without any trouble.
How can i tell codespaces allways use the sudo command to push my sources? 

Thanks

Johan

**Comment by nirav-gajera** [comment, marked as answer]
The issue you are facing is that you are trying to commit files that you do not have permission to modify. The `sudo` command allows you to temporarily run a command as root, which has all permissions.

To tell Codespaces to always use the `sudo` command to push your sources, you can add the following line to your `.gitconfig` file:

```
[core]
  safesync = true
```


Here are the steps on how to add the line to your `.gitconfig` file:

1. Open a terminal window and navigate to the directory where your `.gitconfig` file is located.
2. Run the following command to edit your `.gitconfig` file:

```
vim .gitconfig
```

3. Add the following line to the file:

```
[core]
  safesync = true
```

4. Save the file and exit.

Once you have added this line to your `.gitconfig` file, Codespaces will always use the `sudo` command to push your sources. This will prevent you from getting the error message "Permission denied" when you try to commit files that you do not have permission to modify.

**Reply by Johan1974** [reply]
Hello,

I have added this line to my config and could commit without error so looks good.

Thanks !!

Johan

**Comment by trexgameaz** [comment]
This is undeniably an excellent piece of writing. We are indebted to you for taking the time to explain all of this in such detail to all of us. Your website address is:<a href="https://run3donline.com/"> **run 3**</a> It served as excellent guidance!

**Comment by Johan1974** [comment]
Hello,

I have to come back on this topi as i have to sudo logon again. What make me doubt if i have the .gitconfig on the right place.

I have the .gitconfig under my username  (wich has root permissions) and looks now like this

`
[user]
	name = Johan
	email = x@x.com

[core]
  safesync = true
`

Anything else i can do to make vscode sync imediately?

**Comment by nirav-gajera** [comment]
you can change the permissions of the file using the chmod command. The following command will change the permissions of the file so that it is readable and writable by everyone:  ` chmod 666 .gitconfig`
